File: backend/src/main/java/com/voyagecraft/entity/Notification.java
package com.voyagecraft.entity;

import com.voyagecraft.enums.AlertSeverity;
import com.voyagecraft.enums.NotificationType;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "notifications")
@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
@Builder
public class Notification {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "trip_id")
    private Trip trip;

    @Enumerated(EnumType.STRING)
    @Column(name = "notification_type", nullable = false, length = 30)
    private NotificationType notificationType;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private AlertSeverity severity;

    @Column(nullable = false, length = 300)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String message;

    @Column(length = 500)
    private String actionUrl;

    @Builder.Default
    private Boolean readStatus = false;

    @Builder.Default
    private Boolean dismissed = false;

    private LocalDateTime scheduledFor;

    @CreationTimestamp
    @Column(updatable = false)
    private LocalDateTime createdAt;
}

File: frontend/src/types/notification.ts
export interface NotificationData {
  id: number;
  tripId: number | null;
  tripTitle: string | null;
  notificationType: string;
  severity: string;
  severityIcon: string;
  title: string;
  message: string;
  actionUrl: string;
  readStatus: boolean;
  dismissed: boolean;
  scheduledFor: string | null;
  createdAt: string;
}

export interface NotificationPreference {
  id: number;
  departureReminders: boolean;
  checkinReminders: boolean;
  collaboratorAlerts: boolean;
  budgetAlerts: boolean;
  weatherAlerts: boolean;
  quietHoursEnabled: boolean;
  quietHoursStart: string | null;
  quietHoursEnd: string | null;
  budgetThresholdPercent: number;
  departureReminderHours: number;
}
